internal/buildmanifest/manifest.go

```diff
@@ -0,0 +1,66 @@
+package buildmanifest
+
+import "gopkg.in/yaml.v2"
+
+const (
+	// Filename of a build manifest.
+	Filename = "build.yml"
+	// CurrentVersion os the manifest file format. See Manifest.SDK.
+	CurrentVersion = "2"
+)
+
+// Artifact is a file copied from one path to another during the build.
+type Artifact struct {
+	Path string `yaml:"path"`
+	Dest string `yaml:"dest"`
+}
+
+// Manifest used for the declarative build system for drivers.
+type Manifest struct {
+	// SDK version. Used to track future changes to this file. Current version is '2'.
+	SDK string `yaml:"sdk"`
+
+	Native struct {
+		// Image is a Docker image used as the native driver runtime.
+		Image string `yaml:"image"`
+		// Static is a list of files that will be copied from native driver's source directory
+		// to the final driver image. Note that those files cannot be modified by the build.
+		Static []Artifact `yaml:"static"`
+		// Deps is a list of apt/apk commands executed in the final driver image.
+		// This directive should be avoided since a different Docker image may be used instead of it.
+		Deps  []string `yaml:"deps"`
+		Build struct {
+			// Gopath directive sets a new gopath for the driver build. Only used by Go drivers.
+			Gopath string `yaml:"gopath"`
+			// Image is a Docker image used to build the native driver.
+			Image string `yaml:"image"`
+			// Deps is a list of shell commands to pull native driver dependencies.
+			// Note that those commands are executed before copying the driver files to the
+			// container, so they can be cached. See Run also.
+			Deps []string `yaml:"deps"`
```

Comment:
I find the names here somewhat confusing. "Deps" mostly makes sense—but "Add" seems to be the same idea that we called "Static" above, and "Run" seems like it should more properly be called "Build" (since it is the build instructions).

Also, the distinction between "Add" and "Artifacts" is a little obscured I think: Is it correct that "Add" artifacts are copied from the source tree to the build container, and "Artifact" artifacts are copied from the build container out into the deployment image?  I ask because I think for "Add" we really mean _container_ rather than _image_, whereas for deployment I think we really do mean _image_ (at least on the output side).

Anyway, I feel as if we could probably make the names and the comments a little more clear here.

Reply:
Clarified comments a bit. Is it any better?

The `Add` field is similar to Docker `ADD` directive. At first `Static` was named `Add` as well, but it was confusing, so I changed it to emphasize that those files cannot be changed by the build and are more like static assets.

Reply:
I think both qualify as static assets—in each case the specified files are copied verbatim. The comments do help, but I feel like there isn't a clear distinction such that these should have different names. The fact that one uses the Docker `ADD` mechanism doesn't really matter when writing the config, right?

I guess I was hoping we could settle on one term for static assets and use that throughout the config, assuming I didn't overlook some important distinction.

Reply:
Perhaps more importantly, it'd be nice if we can harmonize the terminology for the different categories of static assets: In the one case, for the build container, in the other for the application image. 

Maybe the name of the target? E.g., `BuildAssets`, `ImageAssets`? I'm open to other ideas too of course. (I know it may seem silly to worry so much about names, but names carry a lot of the weight of the documentation)

Reply:
I'm all in for making the docs as clear as possible. In other cases it will probably be left untouched for months to come :grin:

Regarding `ADD`, indeed it doesn't matter but it may provide a clue for someone familiar with Docker. But `Static` won't.

The only specific field is `Artifacts` - it copies files from build container to the image instead of from source as other similar directives.

`BuildAssets` and `ImageAssets` (or maybe `*Files`?) sounds good to me.

We might also increment the format version to reflect this change to config as well, but it's probably doesn't make that much sense, given an effort that is required.